src/api/todos.ts

```diff
@@ -0,0 +1,75 @@
+import { ErrorType } from '../types/ErrorType';
+import { Filter } from '../types/Filter';
+import { NewTodo } from '../types/NewTodo';
+import { Todo } from '../types/Todo';
+import { client } from '../utils/fetchClient';
+
+export const getTodos = (userId: number): Promise<Todo[]> => {
+  return client.get<Todo[]>(`/todos?userId=${userId}`);
+};
+
+export const addTodo = (userId: number, todo: NewTodo) => {
+  return client.post<Todo>(`/todos?userId=${userId}`, todo);
+};
+
+export const deleteTodo = (todoId: number) => {
+  return client.delete(`/todos/${todoId}`);
+};
+
+export const updateTodoStatus = (
+  userId: number,
+  todoId: number,
+  status: boolean,
+): Promise<Todo> => {
+  return client.patch(`/todos/${todoId}?userId=${userId}`, { completed: status });
+};
+
+export const updateTodoTitle = (
+  userId: number,
+  todoId: number,
+  title: string,
+): Promise<Todo> => {
+  return client.patch(`/todos/${todoId}?userId=${userId}`, { title });
+};
+
+export function generateErrorMessage(errorType: ErrorType) {
+  switch (errorType) {
+    case ErrorType.LOAD:
+      return 'Unable to load todos: server is unavailable';
+    case ErrorType.EMPTY_TITLE:
+      return 'Title can\'t be empty';
+    case ErrorType.DELETE:
+      return 'Unable to add a todo';
+    case ErrorType.ADD:
+      return 'Unable to add a todo';
+    case ErrorType.UPDATE:
+      return 'Unable to update a todo';
+    default:
+      return '';
+  }
+}
```

Comment:
you can add these messages as properties to enum keys, and then call `{ErrorType.UPDATE}` for example

Reply:
It's a bad idea:)